homeassistant/components/homekit_controller/connection.py

```diff
@@ -237,10 +239,8 @@ def async_load_platforms(self):
                 if platform in self.platforms:
                     continue
 
-                self.hass.async_create_task(
-                    self.hass.config_entries.async_forward_entry_setup(
-                        self.config_entry, platform
-                    )
+                await self.hass.config_entries.async_forward_entry_setup(
+                    self.config_entry, platform
                 )
                 self.platforms.add(platform)
```

Comment:
You should add this before awaiting, or else you might forward the entry twice if `async_load_platforms` is called twice  ?

Reply:
That's a good point. I think it's that way round to avoid adding it before knowing the forward was successful - i guess i should try: except: and remove on failure?